### PR TITLE
Fix distill docs typo

### PR DESCRIPTION
**Pull request by AAnoosheh**
# What does this PR do ?

Distillation accidentally mentioned `--nproc-per-node=1` instead of `2`

# Changelog

- Change `--nproc-per-node=1` to `--nproc-per-node=2`

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **Documentation**
  * Updated all example commands in the Llama distillation recipe documentation to ensure accuracy and correctness. The distributed training setup examples now properly reflect current configuration parameters. This helps users following the documentation to successfully configure their training environments with the correct settings needed for reliable and reproducible distillation workflow results.

### CODE REVIEW

File: docs/training/distillation.md [hunk 1]
```diff
@@ -49,15 +49,15 @@ logit_kl_temperature: 2.0
 The simplest way to run knowledge distillation is to use or adapt one of the provided recipe scripts. Here's an example for distilling Llama3.2-3B into Llama3.2-1B:
 
 ```bash
-torchrun --nproc_per_node=1 examples/distillation/llama/distill_llama32_3b-1b.py
+torchrun --nproc_per_node=2 examples/distillation/llama/distill_llama32_3b-1b.py
```
Comment (yaoyu-33) [inline docs/training/distillation.md]: @AAnoosheh  can we update all to `uv run -m torch.distributed.run --nproc_per_node=2`

**Comment by AAnoosheh** [comment]
/ok to test [e24b729](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/3118/commits/e24b7298f35b4245d008f5f9ba14e98fa8550deb)
